cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -78,15 +78,6 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string, discove
 	} else {
 		// If the discovery file config does not contains authentication credentials
 		klog.V(1).Info("[discovery] Discovery file does not contains authentication credentials, using unauthenticated request for validating TLS connection")
-
-		// Create a new kubeconfig object from the discovery file config, with only the server and the CA cert.
-		// NB. We do this in order to not pick up other possible misconfigurations in the clusterinfo file
-		config = kubeconfigutil.CreateBasic(
-			currentCluster.Server,
-			clustername,
```

Comment:
After this is removed, the input param `clustername`  will not be used anyway.

A nit.

Reply:
Yes, this parameter is no longer needed. I don't think the cluster name matters here, `kubelet` always sets the name to `default-cluster` in bootstrap process.
https://github.com/kubernetes/kubernetes/blob/7530baa9a186a08b99fb63060a869344a1f9f4aa/pkg/kubelet/certificate/bootstrap/bootstrap.go#L187

Reply:
The `clustername` parameter has been cleaned since uses can specify it by discovery kubeconfig file.

Reply:
/lgtm